aeon/distances/_minkowski.py

```diff
@@ -0,0 +1,232 @@
+    if w is not None:
+        if x.shape != w.shape:
+            raise ValueError("Weights w must have the same shape as x and y")
+        if np.any(w < 0):
+            raise ValueError("Input weights should be all non-negative")
+        w = w.astype(x.dtype)
+
+    if x.ndim == 1 and y.ndim == 1:
+        return _univariate_minkowski_distance(x, y, p, w)
+    if x.ndim == 2 and y.ndim == 2:
+        return _multivariate_minkowski_distance(x, y, p, w)
+
+    raise ValueError("Inconsistent dimensions.")
+
+
+@njit(cache=True, fastmath=True)
+def _univariate_minkowski_distance(
+    x: np.ndarray, y: np.ndarray, p: float, w: np.ndarray = None
+) -> float:
+    if w is None:
+        dist = np.sum(np.abs(x - y) ** p)
+    else:
+        dist = np.sum(w * (np.abs(x - y) ** p))
+    return dist ** (1.0 / p)
+
+
+@njit(cache=True, fastmath=True)
+def _multivariate_minkowski_distance(
+    x: np.ndarray, y: np.ndarray, p: float, w: np.ndarray = None
+) -> float:
+    dist = 0.0
+    for i in range(x.shape[0]):  # Iterate over rows (time points/instances)
+        diff = np.abs(x[i] - y[i]) ** p
+        if w is None:
+            dist += np.sum(diff)
+        else:
+            dist += np.sum(w[i] * diff)
+
+    return dist ** (1.0 / p)
+
+
+@njit(cache=True, fastmath=True)
+def minkowski_pairwise_distance(
+    X: np.ndarray, y: np.ndarray = None, p: float = 2.0, w: np.ndarray = None
+) -> np.ndarray:
+    """Compute the Minkowski pairwise distance between a set of time series.
+
+    Parameters
+    ----------
+    X : np.ndarray
+        A collection of time series instances, of shape
+        (n_instances, n_channels, n_timepoints) or
+        (n_instances, n_timepoints) or (n_timepoints,).
+    y : np.ndarray, optional
+        A second collection of time series instances, of
+        shape (m_instances, m_channels, m_timepoints) or
+        (m_instances, m_timepoints) or (m_timepoints,).
+        If None, the pairwise distances are calculated within X.
+    p : float, optional
+        The order of the norm of the difference
+        (default is 2.0, which represents the Euclidean distance).
+    w : np.ndarray, optional
+        An array of weights, applied to each pairwise calculation.
+        The weights should match the shape of the time series in X and y.
+
+    Returns
+    -------
+    np.ndarray
+        Minkowski pairwise distance matrix between
+        the instances of X (and y if provided).
+
+    Raises
+    ------
+    ValueError
+        If X (and y if provided) are not 1D, 2D or 3D arrays.
+        If p is less than 1.
+        If w is provided and its shape does not match the instances in X and y.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from aeon.distances import minkowski_pairwise_distance
+    >>> X = np.array([[[1, 2, 3, 4]],[[4, 5, 6, 3]], [[7, 8, 9, 3]]])
+    >>> minkowski_pairwise_distance(X, p=1)
+    array([[ 0. 10. 19.],
+          [10.  0.  9.],
+          [19.  9.  0.]])
```

Comment:
```suggestion
    array([[ 0., 10., 19.],
          [10.,  0.,  9.],
          [19.,  9.,  0.]])
```
The tests get very specific when it comes to doctest output, it can be pretty annoying 🙂